### PR TITLE
perf: Add special case to `Table.add_rows` to increase performance

### PR DESCRIPTION
**Pull request by Marsmaennchen221**
See #606 

### Summary of Changes

perf: Add special case to `Table.add_rows` to increase performance

### CODE REVIEW

File: src/safeds/data/tabular/containers/_table.py [hunk 1]
```diff
@@ -1069,6 +1069,8 @@ def add_rows(self, rows: list[Row] | Table) -> Table:
                         key={val: ix for ix, val in enumerate(self.column_names)}.__getitem__,
                     ),
                 )
+            if self.number_of_rows == 0:
```
Comment (WinPlay02) [inline src/safeds/data/tabular/containers/_table.py:1072]:
```suggestion
            if rows.number_of_rows == 0:
                return self
            if self.number_of_rows == 0:
```

Then we also don't concat if there's nothing to do, the other way round

**Review by WinPlay02 — APPROVED** [review]
Consider the suggestion, otherwise I agree. I'll close my second PR then.

I guess checking the schema will ensure correctness and will prevent surprising the user, that sometimes an `add_rows` operation works, and sometimes it doesn't

**Review by WinPlay02 — APPROVED** [review]
Looks good

**Review by lars-reimann — APPROVED** [review]

**Pull request merged**